### ISSUE TITLE
Race condition in StartSECC.main()

### ISSUE THREAD

**Issue by lategoodbye** [eclipse/risev2g#8]
I noticed a race condition in the main function of [StartSECC](https://github.com/eclipse/risev2g/blob/master/RISE-V2G-SECC/src/main/java/org/eclipse/risev2g/secc/main/StartSECC.java) during init of V2GCommunicationSessionHandlerSECC. If i start the EVCC program on a PC after the UDPServerThread of SECC has been started and before the V2GCommunicationSessionHandlerSECC is initialized then a Null pointer exception occurs because the MessageHandler of the SECC isn't initialized (still null):

```
ERROR [UDPServerThread] V2GCommunicationSessionHandlerSECC: NullPointerException 
occurred while processing SECCDiscoveryReq
java.lang.NullPointerException
        at org.eclipse.risev2g.secc.session.V2GCommunicationSessionHandlerSECC.processSECCDiscoveryReq(V2GCommunicationSessionHandlerSECC.java:127) 
[secc-1.0.0-SNAPSHOT.jar:?]
        at org.eclipse.risev2g.secc.session.V2GCommunicationSessionHandlerSECC.update(V2GCommunicationSessionHandlerSECC.java:68) 
[secc-1.0.0-SNAPSHOT.jar:?]
        at java.util.Observable.notifyObservers(Observable.java:159) [?:1.8.0_121]
        at org.eclipse.risev2g.secc.transportLayer.UDPServer.run(UDPServer.java:100) [secc-1.0.0-SNAPSHOT.jar:?]
```

Test setup:
```
  Role: SECC                       Role: EVCC
[EVAcharge SE] ---- Ethernet ---- [    PC    ]
```

SECC start trace:
```
2017-05-03T17:52:01,429 INFO [main] UDPServer: UDP server initialized at link-local address fe80:0:0:0:201:87ff:fe05:2cc4%eth0 and port 15118
2017-05-03T17:52:14,866 DEBUG [main] TLSServer: TLS server initialized at link-local address fe80:0:0:0:201:87ff:fe05:2cc4%eth0 and port 53922
2017-05-03T17:52:14,993 DEBUG [main] TCPServer: TCP server initialized at link-local address fe80:0:0:0:201:87ff:fe05:2cc4%eth0 and port 54233
2017-05-03T17:52:15,035 DEBUG [TCPServerThread] TCPServer: Waiting for new TCP client connection ...
2017-05-03T17:52:15,050 DEBUG [TLSServerThread] TLSServer: Waiting for new TLS client connection ...
2017-05-03T17:52:15,041 WARN [main] StartSECC: start sessionHandler !!!
2017-05-03T17:52:15,307 WARN [main] V2GCommunicationSessionHandlerSECC: addObserver
2017-05-03T17:52:15,347 WARN [main] V2GCommunicationSessionHandlerSECC: SessionMap
2017-05-03T17:52:15,400 WARN [main] V2GCommunicationSessionHandlerSECC: ConnectionMap
2017-05-03T17:52:15,428 WARN [main] V2GCommunicationSessionHandlerSECC: create MessageHandler
2017-05-03T17:52:15,716 WARN [main] MessageHandler: OpenEXICodec
2017-05-03T17:53:55,710 WARN [main] OpenEXICodec: OpenEXICodec
2017-05-03T17:55:12,776 WARN [main] V2GCommunicationSessionHandlerSECC: setMessageHandler 
```

In order to fix this race condition, i suggest the following SECC init sequence:
1. init all threads
2. init V2GCommunicationSessionHandlerSECC
3. start threads

**Comment by MarcMueltin** [comment]
Hi Stefan, 

thanks for the feedback! I will get into it as soon as I tackle some other issues I have on my list as well. Might take a few weeks due to other tasks I need to take care of and because of vacation time, but stay tuned.

**Comment by lategoodbye** [comment]
@MarcMueltin Should this issue move to V2GClarity/RISE-V2G ?

**Comment by MarcMueltin** [comment]
@lategoodbye This issue has been solved with version 1.0 at V2GClarity/RISE-V2G. Not sure how to transfer an issue though.

**Comment by lategoodbye** [comment]
Thanks
